extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -392,8 +392,10 @@ class MultipleBackendTask extends ReplicateObject {
             // part so the range in the subsequent GET request is undefined.
             return [null];
         }
-        const size = isGCP ?
-            this._getGCPRangeSize(contentLen) : this._getRangeSize(contentLen);
+        let size = this._getRangeSize(contentLen);
+        if (isGCP && (contentLen / size > MPU_GCP_MAX_PARTS)) {
```

Comment:
Although I'm good with the logic, I think it would be cleaner if this logic goes in `_getGCPRangeSize()` (which then would call `_getRangeSize()`)

Reply:
Sounds good! I've updated accordingly and added a unit test that case in `_getRangeSize`.